learn/advanced/pagination.md

```diff
@@ -129,95 +140,124 @@ document.querySelector('#previous_button').onclick = function () { updatePageNum
-  // The amount of results we want to display in each page
-  const hitsPerPage = 10
-
-  // The maximum amount of results we will display
-  const { hits } = await index.search(searchTerm, { limit: 300 })
-
-  // Clear the previous buttons…
-  const pagination = document.querySelector('.pagination')
-  pagination.innerHTML = ''
-
-  // …and create a new button for each search results page
-  for (let page = 0; page < hits.length; page += hitsPerPage) {
-    const numberBtn = document.createElement('button');
-    numberBtn.innerText = page / hitsPerPage
-
-    // When the user clicks on a page number, show that results page
-    numberBtn.onclick = function () { 
-      const pageNumber = parseInt(this.innerText)
-      changePage(pageNumber, hits, hitsPerPage)
-    }
-
-    pagination.appendChild(numberBtn)
+const pageNavigation = document.querySelector('#page-navigation');
+const listContainer = pageNavigation.querySelector('#page-list');
+const results = await index.search(
+  "tarkovsky", 
+  { 
+    hitsPerPage: 25, 
+    page: 1 
   }
+);
 
-  // Display first search results page
-  changePage(1, hits, hitsPerPage)
-}
+const totalPages = results.totalPages;
+const currentPage = results.page;
 
-// Display a page of search results
-function changePage(pageNumber, hits, hitsPerPage) {
-  const offset = (pageNumber - 1) * hitsPerPage
-  const paginatedHits = hits.slice(offset, offset + hitsPerPage)
-  const resultList = document.querySelector('.results')
+for (let i = 0; i < totalPages; i += 1) {
+  const listItem = document.createElement('li');
+  const pageButton = document.createElement('button');
 
-  resultList.innerHTML = ''
+  pageButton.innerHTML = i;
 
-  paginatedHits.map(hit => {
-    const resultItem = document.createElement('div');
-    resultItem.innerText = hit.title
-    resultList.appendChild(resultItem)
-  })
-}
+  if (currentPage === i) {
+    listItem.classList.add("current-page");
+  }
 
-// Get the search bar and retrieve results every time the user makes a new query
-const searchBar = document.querySelector('.searchBar')
-searchBar.onChange(function () { getSearchResults('tarkovsky') })
+  listItem.append(pageButton);
+  listContainer.append(listItem);
+}
 ```
 
-This method provides control and reliability. It also prevents users from paginating too far, which might result in performance gains. However, it limits the number of results your users can see. Additionally, we recommend caution when setting high values for `limit` as it can negatively impact performance.
+#### Adding navigation buttons
 
-::: note
-By default, Meilisearch returns a maximum of 1000 search results. Consult our [index setting reference](/reference/api/settings.md#pagination) if you need to change this.
-:::
+Your users are likely to be more interested in the page immediately after or before the current search results page. Because of this, it is often helpful to add "next" and "previous" buttons to your page list.
```

Comment:
```suggestion
Your users are likely to be more interested in the page immediately after or before the current search results page. Because of this, it is often helpful to add "Next" and "Previous" buttons to your page list.
```